unitTests/ui/DynamicHierarchicalFacetSearchValueRendererTest.ts

```diff
@@ -0,0 +1,265 @@
+  return classNames.map((className, index) => {
+    const childAtIndex = element.childNodes.item(index) as HTMLElement;
+    expectClass(childAtIndex, className);
+    return childAtIndex;
+  });
+}
+
+export function DynamicHierarchicalFacetSearchValueRendererTest() {
+  describe('DynamicHierarchicalFacetSearchValueRenderer', () => {
+    let facetValue: DynamicHierarchicalFacetSearchValue;
+    let facet: IDynamicHierarchicalFacet;
+    let renderer: DynamicHierarchicalFacetSearchValueRenderer;
+
+    function initializeFacet(): IDynamicHierarchicalFacet {
+      facet = mockComponent<DynamicHierarchicalFacet>(DynamicHierarchicalFacet);
+      facet.options = {
+        field: '@title',
+        clearLabel: 'begone'
+      };
+      return facet;
+    }
+
+    function initializeFacetValue(parents: string[] = []): DynamicHierarchicalFacetSearchValue {
+      return setProperties((facetValue = mock<DynamicHierarchicalFacetSearchValue>(DynamicHierarchicalFacetSearchValue)), {
+        displayValue: 'A B C',
+        numberOfResults: 1337,
+        fullPath: [...parents, 'abc']
+      });
+    }
+
+    function initializeRenderer(parents: string[] = []) {
+      return (renderer = new DynamicHierarchicalFacetSearchValueRenderer(initializeFacetValue(parents), initializeFacet()));
+    }
+
+    describe('without any parent', () => {
+      beforeEach(() => {
+        initializeRenderer();
+      });
+
+      describe('when selected', () => {
+        beforeEach(() => {
+          renderer.selectAction();
+        });
+
+        it('calls facet.selectPath with the full path', () => {
+          expect(facet.selectPath).toHaveBeenCalledTimes(1);
+          expect(facet.selectPath).toHaveBeenCalledWith(facetValue.fullPath);
+        });
+
+        it('calls facet.enableFreezeFacetOrderFlag', () => {
+          expect(facet.enableFreezeFacetOrderFlag).toHaveBeenCalledTimes(1);
+        });
+
+        it('calls facet.enablePreventAutoSelectionFlag', () => {
+          expect(facet.enablePreventAutoSelectionFlag).toHaveBeenCalledTimes(1);
+        });
+
+        it('calls facet.scrollToTop', () => {
+          expect(facet.scrollToTop).toHaveBeenCalledTimes(1);
+        });
+
+        it('calls facet.triggerNewQuery with a function that calls facetValue.logSelectActionToAnalytics', () => {
+          expect(facet.triggerNewQuery).toHaveBeenCalledTimes(1);
+          const [logAnalytics] = <[Function]>(facet.triggerNewQuery as jasmine.Spy).calls.mostRecent().args;
+          expect(logAnalytics).toBeTruthy();
+
+          expect(facetValue.logSelectActionToAnalytics).not.toHaveBeenCalled();
+          logAnalytics();
+          expect(facetValue.logSelectActionToAnalytics).toHaveBeenCalledTimes(1);
+        });
+      });
+
+      describe('when rendered', () => {
+        let render: HTMLElement;
+
+        beforeEach(() => {
+          render = renderer.render();
+        });
+
+        it('should have an accessible label', () => {
```

Comment:
I think it's better to add an actual accessibility test (that uses axe core) instead of only checking labels and aria properties.

You don't need to remove this since it's already there, but axe-core would help us know if the properties are valid (versus here we just check they exists, without knowing if they "make sense")

Reply:
Good point, I'll add axe tests.